src/security/core/src/dds_security_serialize.c

```diff
@@ -549,14 +549,14 @@ DDS_Security_Deserialize_string(
     if (dser->remain < sz) {
         return 0;
     }
-
-    if (sz > 0 && (dser->cursor[sz-1] == 0)) {
-       *value = ddsrt_strdup((char *)dser->cursor);
-       dser->cursor += sz;
-       dser->remain -= sz;
-    } else {
-       *value = ddsrt_strdup("");
+    if (sz == 0 || dser->cursor[sz-1] != 0) {
+        return 0;
     }
+
+    ddsrt_free (*value);
+    *value = ddsrt_strdup((char *)dser->cursor);
```

Comment:
I would like to reject the input for that but it comes at the cost of yet another `strlen` and it seems relatively harmless. For Cyclone and anything it does internally, the `strlen` length matters, and for the purposes it is used here it would at worst mean a disagreement on the validity or the interpretation of the data. This could result in failing to communicate, but not on anything worse that I can see.

More generally, given the IDL-to-C mapping, any system that expects to incorporate C code cannot afford to put `\0`s in strings because there is no separate length field ...

So yes, I think it is fine to risk `strlen` potentially returning less than the allocated length.